x-pack/plugins/security/server/authentication/authenticator.test.ts

```diff
@@ -1186,6 +1275,7 @@ describe('Authenticator', () => {
         provider: mockSessVal.provider,
         state: { authorization },
       });
+      expect(auditLogger.log).not.toHaveBeenCalled();
```

Comment:
__question:__ Hmm, does it mean we don't record "user_login" when session is created via `authenticate` method?

If, for example, user enables only PKI or Kerberos authentication without Login Selector then Kibana will create a new session (effectively will log user in) via `authenticate` method.

I just enabled security and audit logs for our PKI integration tests and here is what we have for one of the tests (accesses `/security/account` with client cert). I can see only `http_request` audit event.

```
[2022-02-04T13:32:27.375+01:00][DEBUG][plugins.security.pki.pki] Trying to authenticate user request to /security/account.

[2022-02-04T13:32:27.376+01:00][DEBUG][plugins.security.pki.pki] Trying to authenticate request via peer certificate chain.

[2022-02-04T13:32:27.376+01:00][DEBUG][plugins.security.pki.pki] Self-signed certificate is detected in certificate chain

[2022-02-04T13:32:27.376+01:00][DEBUG][plugins.security.pki.pki] Peer certificate chain: [{"subject":{"CN":"first_client"},"issuer":{"CN":"Elastic Certificate Tool Autogenerated CA"},"issuerCertType":"object","validFrom":"Oct 13 10:47:19 2021 GMT","validTo":"Oct  1 10:47:19 2071 GMT"}, {"subject":{"CN":"Elastic Certificate Tool Autogenerated CA"},"issuer":{"CN":"Elastic Certificate Tool Autogenerated CA"},"issuerCertType":"object","validFrom":"Oct 13 10:15:41 2021 GMT","validTo":"Oct  1 10:15:41 2071 GMT"}]

[2022-02-04T13:32:27.491+01:00][DEBUG][plugins.security.pki.pki] Successfully retrieved access token in exchange to peer certificate chain.

[2022-02-04T13:32:27.491+01:00][DEBUG][plugins.security.session.4Y7FmN/Z8=] Creating a new session.

[2022-02-04T13:32:27.974+01:00][DEBUG][plugins.security.session.4Y7FmN/Z8=] Successfully created a new session.

{"event":{"action":"http_request","category":["web"],"outcome":"unknown"},"http":{"request":{"method":"get"}},"url":{"domain":"localhost","path":"/security/account","port":5620,"scheme":"https"},"user":{"name":"first_client","roles":["kibana_admin"]},"kibana":{"space_id":"default"},"trace":{"id":"2721c9fa-6220-49e6-9695-ec9ae3f9a23a"},"ecs":{"version":"8.0.0"},"@timestamp":"2022-02-04T13:32:27.976+01:00","message":"User is requesting [/security/account] endpoint","log":{"level":"INFO","logger":"plugins.security.audit.ecs"},"process":{"pid":8198},"transaction":{"id":"2de048cadbdb2990"}}

[2022-02-04T13:32:28.024+01:00][DEBUG][plugins.security.pki.pki] Trying to authenticate user request to /internal/security/me

......

```

Reply:
Whoops, I'm glad you caught this, this is exactly why I wanted to add the extra test assertions.

After discussing offline, we agreed that this has grown outside the scope of this PR. I opened a separate issue for us to address this: #124694